### PR TITLE
Add upgrade routine for 0.38.0

### PR DESCRIPTION
**Pull request by VineethReddy02**
Part of: #488

### CODE REVIEW

File: pkg/collector/upgrade/v0_38_0.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	// return if args exist
+	if len(otelcol.Spec.Args) == 0 {
+		return otelcol, nil
+	}
+
+	// Remove otelcol args --log-level, --log-profile, --log-format
+	// are deprecated in reference to https://github.com/open-telemetry/opentelemetry-collector/pull/4213
+	var foundLoggingArgs []string
+	for argKey := range otelcol.Spec.Args {
+		if argKey == "--log-level" || argKey == "--log-profile" || argKey == "--log-format" {
```
Comment (jpkrohling) [inline pkg/collector/upgrade/v0_38_0.go:35]: Can/should we translate that into config changes, to accomplish the same? I believe most, or all of those properties are now available in the collector config itself.

**Review by jpkrohling — APPROVED** [review]

File: pkg/collector/upgrade/v0_38_0.go (hunk 1 above)
Comment (VineethReddy02) [reply pkg/collector/upgrade/v0_38_0.go:35]: config changes? I do not see an option in otel collector configuration which includes args for otel collector container. :) @jpkrohling

Comment (jpkrohling) [reply pkg/collector/upgrade/v0_38_0.go:35]: No, I mean that the args from the collector became options in the collector config: https://github.com/open-telemetry/opentelemetry-collector/pull/4009 . Like this:

```
service:
  telemetry:
    logs:
      level: "DEBUG"
      development: true
      encoding: "json"
```

**Review by jpkrohling — APPROVED** [review]
LGTM.

**Pull request merged**